dbms/src/Storages/KVStore/Decode/PartitionStreams.cpp

```diff
@@ -112,79 +85,187 @@ static DM::WriteResult writeRegionDataToStorage(
 
             if (region.pre_decode_cache->schema_version == schema_version)
             {
-                block = std::move(region.pre_decode_cache->block);
-                need_decode = false;
+                return std::move(region.pre_decode_cache->block);
             }
             else
             {
                 LOG_DEBUG(log, "schema version not equal, try to re-decode region cache into block");
                 region.pre_decode_cache->block.clear();
             }
         }
+        return std::nullopt;
+    }
 
-        /// Read region data as block.
-        Stopwatch watch;
+    const LoggerPtr & log;
+    const Context & context;
+    const TMTContext & tmt;
+    KeyspaceID keyspace_id;
+    TableID table_id;
+    DM::WriteResult write_result = std::nullopt;
+    UInt64 region_decode_cost = -1;
+    UInt64 write_part_cost = -1;
+};
+
+static void inline writeCommittedBlockDataIntoStorage(
+    AtomicReadWriteCtx & rw_ctx,
+    TableStructureLockHolder & lock,
+    ManageableStoragePtr & storage,
+    Block & block)
+{
+    /// Write block into storage.
+    // Release the alter lock so that writing does not block DDL operations
+    TableLockHolder drop_lock;
+    std::tie(std::ignore, drop_lock) = std::move(lock).release();
```

Comment:
```suggestion
        TableLockHolder drop_lock = std::get<1>(std::move(lock).release());
```

Reply:
I think it is better that we drop the lock explicitly here